cpp/src/parquet/stream_writer_test.cc

```diff
@@ -201,26 +261,136 @@ TEST_F(TestStreamWriter, EndRowGroup) {
+  EXPECT_NO_THROW(writer_ << EndRowGroup);
+  EXPECT_NO_THROW(writer_ << EndRowGroup);
+}
+
+TEST_F(TestStreamWriter, SkipColumns) {
+  EXPECT_EQ(0, writer_.SkipColumns(0));
+  EXPECT_THROW(writer_.SkipColumns(2), ParquetException);
+  writer_ << true << std::string("Cannot skip mandatory columns");
+  EXPECT_THROW(writer_.SkipColumns(1), ParquetException);
+  writer_ << 'x' << std::array<char, 4>{'A', 'B', 'C', 'D'} << int8_t(2) << uint16_t(3)
+          << int32_t(4) << uint64_t(5) << 6.0f << 7.0;
+  writer_ << EndRow;
+}
+
+class TestOptionalFields : public ::testing::Test {
+ protected:
+  void SetUp() {
+    writer_ = StreamWriter{ParquetFileWriter::Open(CreateOutputStream(), GetSchema())};
+  }
+
+  void TearDown() { writer_ = StreamWriter{}; }
+
+  std::shared_ptr<schema::GroupNode> GetSchema() {
+    schema::NodeVector fields;
+
+    fields.push_back(schema::PrimitiveNode::Make("bool_field", Repetition::OPTIONAL,
+                                                 Type::BOOLEAN, ConvertedType::NONE));
+
+    fields.push_back(schema::PrimitiveNode::Make("string_field", Repetition::OPTIONAL,
+                                                 Type::BYTE_ARRAY, ConvertedType::UTF8));
+
+    fields.push_back(schema::PrimitiveNode::Make("char_field", Repetition::OPTIONAL,
+                                                 Type::FIXED_LEN_BYTE_ARRAY,
+                                                 ConvertedType::NONE, 1));
+
+    fields.push_back(schema::PrimitiveNode::Make("char[4]_field", Repetition::OPTIONAL,
+                                                 Type::FIXED_LEN_BYTE_ARRAY,
+                                                 ConvertedType::NONE, 4));
+
+    fields.push_back(schema::PrimitiveNode::Make("int8_field", Repetition::OPTIONAL,
+                                                 Type::INT32, ConvertedType::INT_8));
+
+    fields.push_back(schema::PrimitiveNode::Make("uint16_field", Repetition::OPTIONAL,
+                                                 Type::INT32, ConvertedType::UINT_16));
+
+    fields.push_back(schema::PrimitiveNode::Make("int32_field", Repetition::OPTIONAL,
+                                                 Type::INT32, ConvertedType::INT_32));
+
+    fields.push_back(schema::PrimitiveNode::Make("uint64_field", Repetition::OPTIONAL,
+                                                 Type::INT64, ConvertedType::UINT_64));
+
+    fields.push_back(schema::PrimitiveNode::Make("float_field", Repetition::OPTIONAL,
+                                                 Type::FLOAT, ConvertedType::NONE));
+
+    fields.push_back(schema::PrimitiveNode::Make("double_field", Repetition::OPTIONAL,
+                                                 Type::DOUBLE, ConvertedType::NONE));
+
+    return std::static_pointer_cast<schema::GroupNode>(
+        schema::GroupNode::Make("schema", Repetition::REQUIRED, fields));
+  }
+
+  StreamWriter writer_;
+};
+
+TEST_F(TestOptionalFields, Output_operator_with_optional_T) {
```

Comment:
nit: this test name style is unusual in the codebase

Reply:
Ok will change.